app/views.py

```diff
@@ -54,5 +61,52 @@ def poll_state(request):
     else:
         data = 'This is not an AJAX request'
 
-    json_data = json.dumps(data)
-    return HttpResponse(json_data, content_type='application/json')
+
+    try:
+        json_data = json.dumps(data)
+        return HttpResponse(json_data, content_type='application/json')
+    except:
+        return HttpResponse("Finished generating PDF")
+
+
+#initailizes pdf generation from tasks
+def start_pdf_gen(request):
+    if 'job' in request.GET:
+        job_id = request.GET['job']
+        job = AsyncResult(job_id)
+        data = job.result or job.state
+        context = {
+            'data': data,
+            'task_id': job_id,
+        }
+        return render(request, "app/PollState.html", context)
+    elif (request.method == 'POST'):
+            job = generate_pdf.delay(request.queryset)
+            return HttpResponseRedirect(reverse('start_pdf_gen') + '?job=' + job.id)
+    else:
+        return HttpResponseRedirect('/')
+
+#Downloads PDF after task is complete
+def download_pdf(request, task_id):
+
+    task_id = 0
+    try:
+        task_id = request.build_absolute_uri().split("task_id=", 1)[1]                   #builds task id from URL
+    except:
+        return HttpResponseRedirect('/')
+
+    work = AsyncResult(task_id)                                                         #get the work from ID
+    if work.ready():                                                                    #check if task from worker is complete
+        try:
+            result = work.get(timeout=1)                                                #get result of work
+            try:
```

Comment:
Is there a reason there are two try statements? Having one try catch should catch all the problems most of the times.